### PR TITLE
i2c driver changes, gy521 drivers, i2c device consts

### PR DESCRIPTION
**Pull request by Sabramz**
Rewrite in Rust using only linked lists.

### CODE REVIEW

File: driver/include/gy521.hpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+#include "i2c.hpp"
+
+class gy521 {
+public:
+	/**
+	 * @brief Initializer
+	 *
+	 * @param[in] chip - libgpiod GPIO chip object.
+	 * @param[in] int_pin - Interrupt libgpiod pin number.
```
Comment (vladaviedov) [inline driver/include/gy521.hpp:24]: Missing argument comments

File: driver/src/gy521.cpp [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+ * @brief Drivers for the GY-521 6 axis accelerometer.
+ */
+
+#include "gy521.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+#include "defines.hpp"
+#include "device.hpp"
```
Comment (vladaviedov) [inline driver/src/gy521.cpp]: necessary?

**Review by vladaviedov — APPROVED** [review]
Good, will merge once tested

File: driver/src/gy521.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ */
+
+#include "gy521.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+#include "defines.hpp"
+
+constexpr uint8_t REG_INTERRUPT_STATUS = 0x3A;
```
Comment (vladaviedov) [inline driver/src/gy521.cpp:13]: Make static

**Comment by Sabramz** [comment]
This branch is now a part of rc552 branch.

**Comment by vladaviedov** [comment]
> This branch is now a part of rc552 branch.

Bad practice

**Pull request closed without merging**